### ISSUE TITLE
Bug in task_set.py

### ISSUE THREAD

**Issue by HugoChaugier** [Continvvm/continuum#112]
Hi, I just noticed that there is, I think, a mistake in task_set.py, line 174:
x, y, t = self._prepare(x, y, y)    
instead of:
 x, y, t = self._prepare(x, y, t).

I had a bug because of that after I reinstalled continuum today, the modification fixed it.

**Comment by TLESORT** [comment]
hmm, it looks like a mistake indeed
Can you share the scenarios that you were using to get a bug ?

**Comment by HugoChaugier** [comment]
Something like:
ClassIncremental( continuum.datasets.MNIST('datasets', download=False, train=True), initial_increment = 6, increment = 2)

**Comment by TLESORT** [comment]
I do not get any error, with this scenario, can you describe a bit more your bug?

**Comment by HugoChaugier** [comment]
It's just that I was retrieving the task id that way:
```
for batch_id, (x, y, _t) in enumerate(train_loader):
                task_id = _t[0].item()
```
and used it as an array index but since with the mistake `_t` was the label tensor instead of the task id tensor the index was wrong.

**Comment by TLESORT** [comment]
Ok thanks, we will fix this asap :)

**Comment by arthurdouillard** [comment]
@HugoChaugier sorry for the bug, @TLESORT has fixed it and I've added a test for that.

You can update your continuum to 1.0.11 where the bug has been fixed :)
